vector/src/main/java/im/vector/riotx/features/home/room/detail/RoomDetailViewModel.kt

```diff
@@ -470,7 +471,20 @@ class RoomDetailViewModel @AssistedInject constructor(@Assisted initialState: Ro
                     type = ContentAttachmentData.Type.values()[it.mediaType]
             )
         }
-        room.sendMedias(attachments)
+
+        val homeServerCapabilities = session.getHomeServerCapabilities()
+
+        val maxUploadFileSize = homeServerCapabilities.maxUploadFileSize
+
+        if (maxUploadFileSize == HomeServerCapabilities.MAX_UPLOAD_FILE_SIZE_UNKNOWN) {
+            // Unknown limitation
+            room.sendMedias(attachments)
+        } else {
+            attachments.find { it.size > maxUploadFileSize }
```

Comment:
This?
```
attachments.find { it.size > maxUploadFileSize }?.also {
    _fileTooBigEvent.postValue(LiveEvent(FileTooBigError(it.name ?: it.path, it.size, maxUploadFileSize)))
    room.sendMedias(attachments)
}
```

Reply:
I do not want to send the attachments if there is a too large file.
Rewritten with `when` for more clarity